error.go

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 The Ebitengine Authors
+
+package purego
+
+// Dlerror represents an error value returned from purego
```

Comment:
This error isn't a generic purego error. It only occurs when using dlopen, dlsym, or dlclose. I think the doc should express that